kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java

```diff
@@ -472,6 +473,12 @@ public void createContainer(String id, KieContainerInstance kieContainerInstance
 
     @Override
     public boolean isUpdateContainerAllowed(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
+
+        // Allowing container updates for SNAPSHOTS
+        if(KieServerUtils.isSnapshot(kieContainerInstance.getKieContainer().getReleaseId())) {
```

Comment:
shouldn't we check the mode here as well? or we rely on the mechanism that snapshots can be deployed to dev mode servers only?

Reply:
@mswiderski I was trying to keep extensions not aware of which mode the kie-server server is running. IMO (and maybe I'm wrong) checking on container creation and container start should be enough. If the container exists it can be updated and then the extension should decide based on it's own parameters. WDYT?

Reply:
+!, let's keep it as is